docs/cli/index.md

```diff
@@ -32,3 +32,226 @@ Workflow, show Workflow History, and Signal Workflow).
+
+```bash
+temporal server start-dev --db-filename my_test.db
+```
+
+### Enable or disable Temporal UI
+
+By default, the Temporal UI is started with Temporal CLI. The UI can be disabled via a runtime flag:
+
+```bash
+temporal server start-dev --headless
+```
+
+To build without static UI assets, use the `headless` build tag when running `go build`.
+
+<!--TODO: add go example -->
+
+### Dynamic configuration
+
+Advanced configuration of the Temporal CLI requires the use of a dynamic configuration file.
+This file is created outside of the Temporal CLI; it is usually located with the service's config files.
+
+Dynamic configuration values can also be set via `--dynamic-config-value KEY=JSON_VALUE`.
+For example, to disable the search attribute cache, run:
+
+```bash
+temporal server start-dev --dynamic-config-value system.forceSearchAttributesCacheRefreshOnRead=true
+```
+
+This setting makes created search attributes immediately available for use.
+
+## Auto-completion
+
+The Temporal CLI has the capability to auto-complete commands.
+
+Running `temporal completion SHELL` will output the related completion SHELL code.
+
+### zsh auto-completion
+
+<!-- TODO: add more information about zsh to make comparable to bash section -->
+
+Add the following code snippet to your `~/.zshrc` file:
+
+```sh
+source <(temporal completion zsh)
+```
+
+If you're running auto-completion from the terminal, run the command below:
+
+```sh
+echo 'source <(temporal completion zsh)' >> ~/.zshrc
+```
+
+After setting the variable, run:
+
+`source ~/.zshrc`.
+
+### Bash auto-completion
+
+Bash auto-completion relies on `bash-completion`.
+
+Install the software with the steps provided [here](https://github.com/scop/bash-completion#installation), or use your preferred package manager on your operating system.
+
+#### Linux installation
+
+Use any of the following package managers to install `bash-completion`:
+`apt install bash-completion`
+`pacman -S bash-completion`
+`yum install bash-completion`
+
+Verify that `bash-completion` is installed by running `type _init_completion`.
+
+To install the software on Alpine Linux, run:
+
+```bash
+apk update
+apk add bash-completion
+source /etc/profile.d/bash_completion.sh
+```
+
+Finally, enable completion for Temporal by adding the following code to your bash file:
+
+```
+echo 'source <(temporal completion bash)' >>~/.bashrc
+source ~/.bashrc
+```
+
+#### macOS installation
+
+Install `bash-completion` through Homebrew:
+`brew install bash-completion@2`
+
+Add the provided code snippet to your `~/.bashrc` file:
+
+```sh
+[[ -r "/usr/local/etc/profile.d/bash_completion.sh" ]] && . "/usr/local/etc/profile.d/bash_completion.sh"
+```
+
+Verify that `bash-completion` is installed by running `type _init_completion`.
+
+Enable completion for Temporal by adding the following code to your bash file:
+
+```
+echo 'source <(temporal completion bash)' >> ~/.bashrc
+source ~/.bashrc
+```
+
+## Development
```

Comment:
this is for CLI development and unnecessary here, can be confusing for external users